edgexfoundry/app-functions-sdk-go
# fix: Make deep copy of MessageBus Optional map so not to leak password

fixes #1108

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:


## Testing Instructions
Run secure EdgeX Stack and stop App Rules Engine
Use this branch to build ASC
Run the following from command-line 
```
 sudo EDGEX_SECURITY_SECRET_STORE=true ./app-service-configurable -p=rules-engine
```
Hit `localhost:59701/api/v2/config` from PostMan
Verify Password not in Trigger Optional Config section
```
            "EdgexMessageBus": {
                "SubscribeHost": {
                    "Host": "edgex-redis",
                    "Port": 6379,
                    "Protocol": "redis",
                    "SubscribeTopics": "edgex/events/#"
                },
                "PublishHost": {
                    "Host": "edgex-redis",
                    "Port": 6379,
                    "Protocol": "redis",
                    "PublishTopic": "rules-events"
                },
                "Type": "redis",
                "Optional": {
                    "AutoReconnect": "true",
                    "ClientId": "app-rules-engine",
                    "ConnectTimeout": "5",
                    "KeepAlive": "10",
                    "Password": "XY+ciscYHUioderBc4u4quKxS/lQGxhbg4zdgMhEeUL5",
                    "Qos": "0",
                    "Retained": "false",
                    "SkipCertVerify": "false",
                    "Username": "redis5",
                    "authmode": "usernamepassword",
                    "secretname": "redisdb"
                }
            },
            "ExternalMqtt": {
                "Url": "",
                "SubscribeTopics": "",
                "PublishTopic": "",
                "ClientId": "",
                "ConnectTimeout": "",
                "AutoReconnect": false,
                "KeepAlive": 0,
                "QoS": 0,
                "Retain": false,
                "SkipCertVerify": false,
                "SecretPath": "",
                "AuthMode": "",
                "RetryDuration": 0,
                "RetryInterval": 0
            }

```
## New Dependency Instructions (If applicable)
